pulsar-broker/src/main/java/org/apache/pulsar/broker/service/persistent/PersistentTopic.java

```diff
@@ -1429,7 +1430,14 @@ private CompletableFuture<Void> delete(boolean failIfHasSubscriptions,
                 }
                 FutureUtil.waitForAll(futures).thenRunAsync(() -> {
                     closeClientFuture.complete(null);
-                }, getOrderedExecutor()).exceptionally(ex -> {
+                }, command -> {
+                    try {
+                        getOrderedExecutor().execute(command);
```

Comment:
This executor will only be closed when the broker is shutdown, so we can skip the closing of this topic here

Reply:
Sure, but that would require more logic. This would address the current retry loop.